gradio/processing_utils.py

```diff
@@ -518,3 +522,61 @@ def _scale(a, n, m, copy=True):
 
 def strip_invalid_filename_characters(filename: str) -> str:
     return "".join([char for char in filename if char.isalnum() or char in "._- "])
+
+
+def ffmpeg_installed() -> bool:
+    return shutil.which("ffmpeg") is not None
+
+
+def video_is_playable(video_filepath: str) -> bool:
+    """Determines if a video is playable in the browser.
+
+    A video is playable if it has a playable container and codec.
+        .mp4 -> h264
+        .webm -> vp9
+        .ogg -> theora
+    """
+    try:
+        container = pathlib.Path(video_filepath).suffix.lower()
+        output = subprocess.run(
+            [
+                "ffprobe",
+                "-show_format",
+                "-show_streams",
+                "-select_streams",
+                "v",
+                "-print_format",
+                "json",
+                video_filepath,
+            ],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+        )
+        output = json.loads(output.stdout)
+        video_codec = output["streams"][0]["codec_name"]
+        return (container, video_codec) in [
+            (".mp4", "h264"),
+            (".ogg", "theora"),
+            (".webm", "vp9"),
+        ]
+    # If anything goes wrong, assume the video can be played to not convert downstream
+    except Exception:
+        return True
+
+
+def convert_video_to_playable_mp4(video_path: str) -> str:
+    """Convert the video to mp4. If something goes wrong return the original video."""
+    try:
+        output_path = pathlib.Path(video_path).with_suffix(".mp4")
+        with tempfile.NamedTemporaryFile() as tmp_file:
+            shutil.copy2(video_path, tmp_file.name)
+            # ffmpeg will automatically use h264 codec (playable in browser) when converting to mp4
+            ff = FFmpeg(
+                inputs={str(tmp_file.name): None},
+                outputs={str(output_path): None},
+                global_options="-y -loglevel quiet",
+            )
+            ff.run()
+    except:
```

Comment:
Same nit: can we narrow the scope to `ffmpeg.Error`

Reply:
Yea good call!

Reply:
I got a ```video_codec= output["streams"][0]["codec_name"]
IndexError: list index out of range
 ``` 
The narrowed scope of the exception broke my application, it worked when I reverted back to `except Exception:`